fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -1175,6 +1185,54 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             !processIntent(intent)
     }
 
+    private fun showFullscreenMessageSurveyIfNeeded(context: Context) {
+        val messagingStorage = context.components.analytics.messagingStorage
+        val messages = runBlockingIncrement { messagingStorage.getMessages() }
+        val nextMessage =
+            messagingStorage.getNextMessage(FenixMessageSurfaceId.SURVEY, messages)
+                ?: return
+
+        if (nextMessage.id == "fullscreen-viewpoint-invitation-message") {
+            val currentBootUniqueIdentifier = BootUtils.getBootIdentifier(context)
+            val fenixNimbusMessagingController = FenixNimbusMessagingController(messagingStorage)
+
+            val researchSurfaceDialogFragment = ResearchSurfaceDialogFragment.newInstance(
+                keyMessageText = nextMessage.data.text,
+                keyAcceptButtonText = nextMessage.data.buttonLabel,
+                keyDismissButtonText = null,
+            )
+
+            researchSurfaceDialogFragment.onAccept = {
+                val actionUri = fenixNimbusMessagingController.processMessageActionToUri(nextMessage)
+                processIntent(Intent(Intent.ACTION_VIEW, actionUri))
+
+                components.appStore.dispatch(AppAction.MessagingAction.MessageClicked(nextMessage))
+            }
+
+            researchSurfaceDialogFragment.onDismiss = {
+                components.appStore.dispatch(AppAction.MessagingAction.MessageDismissed(nextMessage))
+            }
+
+            lifecycleScope.launch(Main) {
+                researchSurfaceDialogFragment.showNow(
+                    supportFragmentManager,
+                    ResearchSurfaceDialogFragment.FRAGMENT_TAG,
+                )
+            }
+
+            // Update message as displayed.
+            val updatedMessage =
```

Comment:
I am little confused here. Why are we calling `updateMessageAsDisplayed` here when it is automatically called in `onMessageDisplayed` below? This will cause the metadata `displayCount` to be incremented by 2. Not that it matters since the `maxDisplayCount` = 1

Reply:
Let me double check.  Thanks for point this out.

Reply:
Confirmed that this is not an issue.  Thanks for the help @sarah541